assets/content/cookbook/Abstract types/working-around-iterator-limitations-as-generic-type-parameter.md

```diff
@@ -0,0 +1,37 @@
+[tags]: / "abstract-type,type-params,iterator"
+
+# Working Around Iterators as a Generic Type Parameter
+
+[Iterators](https://haxe.org/manual/lf-iterators.html) are a helpful Haxe structure. They allow you to build a for loop, but sometimes you want to do something a bit more complex, like iterating over a list of items over time, which excludes the possibility of using a for loop. Ideally you would build a generator function, but Haxe does not have these out of the box. There are libraries that allow you to do this, though, like [moon-core](https://lib.haxe.org/p/moon-core/1.0.0/).
+
+However, Iterator is a structure, which is a limitation when you want to use it as the parameter for a generic type. For example, haxe.ds.[GenericStack](http://api.haxe.org/haxe/ds/GenericStack.html)<Iterator<Int>> is an interesting piece of code. It will compile for an html5 target, given Javascript's extreme type flexibility, but it will not compile for other targets, including Flash. The compiler will display the following error:
```

Comment:
Well, I'm working with javascript, and I found the error first when trying to compile to cpp. I thought that flash being similar to js could be more flexible and accept the code, but it didn't. I guess it's just the way I managed to find the error and figure out a solution.